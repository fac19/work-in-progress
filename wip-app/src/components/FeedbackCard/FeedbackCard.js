import React from "react";
import { FeedbackText, FeedbackTag } from "./FeedbackCard.style";

const FeedbackCard = ({ feedback_text, feedback_tag }) => {
  // <p>Posted by: </p>
  return (
    <article>
      <FeedbackText>Comment: {feedback_text}</FeedbackText>
      <FeedbackTag>Tag: {feedback_tag}</FeedbackTag>
    </article>
  );
};

export default FeedbackCard;
