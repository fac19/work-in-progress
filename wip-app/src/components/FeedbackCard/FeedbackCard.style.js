import styled from "styled-components";

const FeedbackText = styled.p`
  margin: 0.5rem;
  margin-bottom: 2rem;
  font-weight: 300;
`;

const FeedbackTag = styled.p`
  margin-top: 0.55rem;
  margin-bottom: 1rem;
  font-weight: 500;
`;

const Comment = styled.div`
  display: flex;
  flex-direction: column;
  border-top: 0.5px solid lightgrey;
  border-bottom: 0.5px solid lightgrey;
  margin-top: 1rem;
  margin-left: 1rem;
  margin-right: 1rem;
`;

const FeedbackUser = styled.h3`
  margin: 0.5rem;
`;

const Step = styled.div`
  display: flex;
  flex-direction: column;
`;

const FeedbackForm = styled.form`
  display: flex;
  flex-direction: row;
  width: 80%;
  justify-content: center;
`;

const FeedbackInput = styled.input`
  width: 90%;
  height: 2rem;
  border-radius: 3px;
  border: 0.5px solid black;
`;

const FeedbackButton = styled.button`
  margin-left: 0.3rem;
  width: 2rem;
  background-color: #2096ee;
  color: white;
  font-size: 1.5rem;
  border: 0.5px solid black;
  border-radius: 3px;
`;

const FeedbackArticle = styled.article`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80%;
  margin: 0 auto;
  margin: 1em 0;
  padding: 3em;
  border-radius: 3em;
  text-align: center;
  box-shadow: 0 0.5em 1em 0 rgba(0, 0, 0, 0.3);
  transition: 0.3s;
  &:hover {
    box-shadow: 0 1em 2em 0 rgba(0, 0, 0, 0.3);
  }
`;

const FeedbackHeader = styled.header`
  border-bottom: 1px solid lightgrey;
  width: 80%;
  display: flex;
  flex-direction: column;
  border-bottom: 0.5px solid lightgrey;
  margin: 1rem;
`;

const FeedbackList = styled.ul`
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 0;
  margin-bottom: 0.1rem;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
`;

export {
  FeedbackText,
  FeedbackTag,
  Comment,
  FeedbackUser,
  Step,
  FeedbackForm,
  FeedbackInput,
  FeedbackButton,
  FeedbackArticle,
  FeedbackHeader,
  FeedbackList,
};
