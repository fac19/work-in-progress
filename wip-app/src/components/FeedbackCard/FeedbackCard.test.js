import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import FeedbackCard from "./FeedbackCard"

test("Jest is working", () => {
  expect(true).toBeTruthy()
})
