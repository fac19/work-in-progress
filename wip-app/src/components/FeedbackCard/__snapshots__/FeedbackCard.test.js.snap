// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<article
  className="sc-AxmLO itYlbA"
>
  <div
    className="sc-AxhUy gfXgDx"
  >
    <header
      className="sc-fzozJi yCEqO"
    >
      <h3
        className="sc-AxhCb klWEsU"
      />
      <p
        className="sc-AxjAm hijcPm"
      />
    </header>
    <img
      alt=""
      className="sc-fzplWN jHAIBM"
    />
  </div>
  <form
    className="sc-AxgMl iNvcfR"
  >
    <label
      htmlFor="feedback"
    />
    <input
      className="sc-AxheI gBxvGf"
      id="tag"
      name="feedback"
      placeholder="Feedback..."
      type="text"
    />
    <button
      className="sc-Axmtr kVnIK"
      type="submit"
    >
      +
    </button>
  </form>
  <div
    className="sc-AxiKw ioNHUL"
  >
    <ul
      className="sc-fzoLsD dXgjRU"
    >
      <h3
        className="sc-AxhCb klWEsU"
      >
         
      </h3>
      <p
        className="sc-AxirZ dJYxqb"
      >
         
      </p>
    </ul>
    <p
      className="sc-AxjAm hijcPm"
    >
       
    </p>
  </div>
  <div
    className="sc-AxhUy gfXgDx"
  >
    <header
      className="sc-fzozJi yCEqO"
    >
      <h3
        className="sc-AxhCb klWEsU"
      />
      <p
        className="sc-AxjAm hijcPm"
      />
    </header>
    <img
      alt=""
      className="sc-fzplWN jHAIBM"
    />
  </div>
  <form
    className="sc-AxgMl iNvcfR"
  >
    <label
      htmlFor="feedback"
    />
    <input
      className="sc-AxheI gBxvGf"
      id="tag"
      name="feedback"
      placeholder="Feedback..."
      type="text"
    />
    <button
      className="sc-Axmtr kVnIK"
      type="submit"
    >
      +
    </button>
  </form>
  <div
    className="sc-AxiKw ioNHUL"
  >
    <ul
      className="sc-fzoLsD dXgjRU"
    >
      <h3
        className="sc-AxhCb klWEsU"
      >
         
      </h3>
      <p
        className="sc-AxirZ dJYxqb"
      >
         
      </p>
    </ul>
    <p
      className="sc-AxjAm hijcPm"
    >
       
    </p>
  </div>
</article>
`;
